src/core/shader.js

```diff
@@ -50,11 +50,16 @@ Shader.prototype = {
    * Called during shader initialization and is only run once.
    */
   init: function (data) {
+    if (this.raw) {
```

Comment:
Not sure this message is very valuable. Perhaps better making a note in the docs. I'm also happy to rip the band-aid and perhaps starting to say that WebGL1 is deprecated and WebGL2 is the way to go.

Reply:
This definitely has to be added to docs. But i think keeping such warning just for a few versions can save some headache to those who still using old shaders and has to migrate.

Reply:
Not sure warning is super descriptive for the avg. dev. It is extra code to maintain and eventually have to remember to remove when obsolete.

What we usually do is to add `deprecations` in the release notes for those migrating https://github.com/aframevr/aframe/releases That's evergreen info.

Reply:
The warning isn't particular useful as it will always be printed. It might even confuse people seeing the warning when everything is correct. The fact that a big error is printed by Three.js down the line is probably enough. Anyone writing their own raw shaders will either understand the error or look it up and find the relevant Three.js and A-Frame issues. Those using older shaders from others will most likely open an issue about _that_ error message anyway.

Documenting this change and mentioning it in the release notes is key, as that will help anyone updating and can easily be linked/pointed to when people encounter this.